src/bank.rs

```diff
@@ -594,7 +555,7 @@ impl Bank {
         {
             let tx = &entry1.transactions[0];
             let instruction = tx.instruction();
-            let deposit = if let Instruction::NewContract(contract) = instruction {
+            let deposit = if let Some(Instruction::NewContract(contract)) = instruction {
```

Comment:
Couldn't we pass `tx` through `process_transaction()` here instead of hard coding a Budget Contract instruction?

Reply:
@mvines maybe, not sure.  its outside the scope of moving budget as a separate contract